### DISCUSSION TITLE
Labels not saving

### DISCUSSION THREAD

**Discussion by clee758** [talmolab/sleap#1767] [Help!]
I am labelling a large dataset (3600 frames), but although I am constantly saving my labels, when I reopen my project, some of the labels are gone/haven't saved. Sometimes, there will be frames that have been saved when I reopen my project a first time, but when I reopen the project a second time, they are gone.

Also, for one video, I kept track of the frames I labelled (50 frames), but when I reopened the project it said there were another 100 frames labelled.

I've never had these issues before with when I was training smaller numbers of frames (<3200). What is going on/are there precautions I can take to make sure this doesn't keep happening?

**Comment by talmo** [comment]
Hi @clee758,

This sounds like a concerning issue which definitely should not happen! Do you mind [sharing the labels](https://forms.gle/KebBPdjoHw2WMKmV8) with us so we can investigate?

Thanks!

Talmo

**Reply by clee758** [reply]
Hi, thank you so much for your quick response! I just shared the labels.

**Comment by eberrigan** [comment]
To mitigate the situation, can you save versions of your labels before and after labeling sessions? Also, please send us the output in the terminal when you are saving. Thanks!

Elizabeth

**Reply by clee758** [reply]
Hi, will do thank you! However, I closed the terminal earlier so I'm not able to share the output

**Reply by eberrigan** [reply]
We recommend saving labels with a versioning system like "labels.v000.slp", "labels.v001.slp" and so on, so that changes in labels can be tracked easily and if something goes wrong it is easy to backtrack. 

Best,

Elizabeth

**Reply by clee758** [reply]
Sounds great, thank you!
